executor/src/witgen/generator.rs

```diff
@@ -50,7 +50,7 @@ impl<'a, T: FieldElement> CompletableIdentities<'a, T> {
 
 pub struct Generator<'a, T: FieldElement> {
     /// The witness columns belonging to this machine
-    witnesses: BTreeSet<PolyID>,
+    witnesses: HashSet<PolyID>,
```

Comment:
can this cause any output nondeterminism?

Reply:
No, it's only used to restrict the columns that are printed, and to implement the `is_relevant_witness()` function, which just checks whether is a function is contained in the set.